pkg/library/gatewayapi/utils.go

```diff
@@ -165,3 +165,23 @@ func IsGatewayAPIInstalled(restMapper meta.RESTMapper) (bool, error) {
 
 	return false, err
 }
+
+// GetGatewayParentRefsFromRoute returns the list of parentRefs that are Gateway typed
+func GetGatewayParentRefsFromRoute(route *gatewayapiv1.HTTPRoute) []gatewayapiv1.ParentReference {
+	if route == nil {
+		return nil
+	}
+	return utils.Filter(route.Spec.ParentRefs, IsParentGateway)
```

Comment:
I do not see any reason to change when `utils.Filter()` does what it is expected to do. Anything I could miss?

Reply:
`github.com/samber/lo` is immensely popular and IMO very good at this kind of generic widely used functions. Wherever we can leverage something like this from the community – provided it suits our needs of course – instead of keeping our own home-made version of the same thing, I personally would be in favour of.

At the same time, I don't want to create a fuss. I understand this is not the main goal of this PR. Just something to think about.